src/bindgen/ir/enumeration.rs

```diff
@@ -764,6 +776,56 @@ impl Source for Enum {
 
             write!(out, "return {};", stream);
             out.close_brace(false);
+
+            if is_tagged {
+                // For tagged enums, this emits the serializer function for
+                // the top-level enum or struct.
+                out.new_line();
+                out.new_line();
+                write!(
+                    out,
+                    "friend std::ostream& operator<<(std::ostream& {}, const {}& {})",
+                    stream,
+                    self.export_name(),
+                    instance,
+                );
+
+                out.open_brace();
+                write!(out, "switch ({}.tag)", instance);
+                out.open_brace();
+                let vec: Vec<_> = self
+                    .variants
+                    .iter()
+                    .map(|x| {
+                        if let VariantBody::Body { ref name, .. } = x.body {
+                            format!(
+                                "case {}::{}::{}: {} << {}.{}; break;",
+                                self.export_name(),
```

Comment:
Do we need the export_name() here? we don't generally use it in other derived code I think.

Reply:
(Assuming this comment goes with the other comment on the expectations file, where the generated code is `F::Tag::Foo` instead of `Tag::Foo` - in which case, yes, I needed this to satisfy the compiler)